mpi/docs/mpi_neighbor_allgather/data.json

```diff
@@ -0,0 +1,66 @@
+{
+    "Type":"Documentation",
+    "Technology":"MPI",
+    "Name":"MPI_Neighbor_allgather",
+    "DirectoryName":"mpi_neighbor_allgather",
+    "SharedDescription":true,
+    "Description":"MPI_Neighbor_allgather is a collective representing an operation that can be used on the topologies MPI_DIST_GRAPH_CREATE, MPI_GRAPH_CREATE, and MPI_CART_CREATE. ",
```

Comment:
MPI_Neighbor_allgather is a neighbourhood collective operation that can be used on virtual topologies. It can be used on Cartesian virtual topologies, created with MPI_Cart_create; and graph virtual topologies created with MPI_Graph_create, MPI_Dist_graph_create or MPI_Dist_graph_create_adjacent. Variants include MPI_Neighbor_allgatherv. For the non-blocking version, refer to MPI_Ineighbor_allgather.